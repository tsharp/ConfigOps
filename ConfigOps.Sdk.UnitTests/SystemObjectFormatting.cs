using ConfigOps.Core.Serialization;
using ConfigOps.Sdk.Dto;
using Snapshooter.Xunit;

namespace ConfigOps.Sdk.UnitTests
{
    public class SystemObjectFormatting
    {
        [Fact]
        public void Validate_SystemObject_YamlFormat()
        {
            SystemObject systemObject = new SystemObject()
            {
                ApiVersion = "v1",
                Kind = "ServiceConfig",
                Metadata = new SystemObjectMetadata()
                {
                    Name = "service-a",
                    Namespace = "default",
                    Labels = new Dictionary<string, string>()
                    {
                        { "geo", "us" },
                        { "region", "westus" },
                        { "env", "prod" }
                    }
                }
            };

            systemObject
                .SerializeToYaml()
                .MatchSnapshot();
        }
    }
}