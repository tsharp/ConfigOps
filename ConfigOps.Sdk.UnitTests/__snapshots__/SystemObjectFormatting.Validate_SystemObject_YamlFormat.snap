apiVersion: v1
kind: ServiceConfig
metadata:
  name: service-a
  namespace: default
  labels:
    geo: us
    region: westus
    env: prod
